source/AI.cpp

```diff
@@ -3131,6 +3209,34 @@ double AI::RendezvousTime(const Point &p, const Point &v, double vp)
 
 
 
+void AI::PlayerTargetAsteroid(Ship &ship, const PlayerInfo &player, Command &activeCommands)
```

Comment:
This doesn't need to be player-specific, e.g.
```c++
AI::TargetAsteroid(Ship &ship, Command &commands)
```
Use a non-void return value (e.g. `bool`, `const Body *`, etc.) to handle the player-specific bits at the call site.

Reply:
Hi, this is the first time I've developed C++ (I'm learning as I go).

> Use a non-void return value (e.g. bool, const Body *, etc.) to handle the player-specific bits at the call site.

I understand in general what you man through knowing many other programming languages.  For this specific request can you be a little more specific.  I do not understand in the context of what I need to change.

Reply:
Any place you currently call this method, you can use the return value to know what the method did. If at that call site it makes sense to do something else, you can do it there.

For example, if you return a pointer to the asteroid that was targeted, you could use that to share the target with the relevant fleet.

Splitting the player-aware bits from this method mean it can be used for ships that are not the player flagship.

Reply:
Okay, I will attempt to update it with this in mind.